### PR TITLE
frontend: compass calibration: add more calibration information and different calibration levels

### PR DESCRIPTION
**Pull request by Williangalvani**
image
image
image

### CODE REVIEW

File: core/frontend/src/components/vehiclesetup/configuration/compass/ArdupilotMavlinkCompassSetup.vue [hunk 1]
```diff
@@ -309,15 +325,64 @@ export default Vue.extend({
           (name) => autopilot_data.parameter(name),
         )
         if (offset_params.includes(undefined) || diagonal_params.includes(undefined)) {
-          results[compass.param] = false
+          results[compass.param] = {
+            calibration: CalibrationType.UNKNOWN,
+            color: 'red',
+            alert: 'error',
+            description: 'Unable to get calibration status, something went wrong.',
```
Comment (ES-Alexander) [inline core/frontend/src/components/vehiclesetup/configuration/compass/ArdupilotMavlinkCompassSetup.vue:332]:
```suggestion
            description: 'Unable to get calibration status - something went wrong.',
```
Should be either a dash, semi-colon, or full-stop+new sentence - comma doesn't really work there.

File: core/frontend/src/components/vehiclesetup/configuration/compass/ArdupilotMavlinkCompassSetup.vue [hunk 2]
```diff
@@ -309,15 +325,64 @@ export default Vue.extend({
...
         }
-        const scale_param_name = `COMPASS_SCALE${compass_number}`
-        const scale_param = autopilot_data.parameter(scale_param_name)
         const is_at_default_offsets = offset_params.every((param) => param?.value === 0.0)
+        if (is_at_default_offsets) {
+          results[compass.param] = {
+            calibration: CalibrationType.NOT_CALIBRATED,
+            color: 'red',
+            alert: 'error',
+            description: 'The compass is not calibrated, please calibrate it.',
```
Comment (ES-Alexander) [inline core/frontend/src/components/vehiclesetup/configuration/compass/ArdupilotMavlinkCompassSetup.vue:343]:
```suggestion
            description: 'The compass is not calibrated - please calibrate it.',
```

File: core/frontend/src/components/vehiclesetup/configuration/compass/ArdupilotMavlinkCompassSetup.vue [hunk 3]
```diff
@@ -309,15 +325,64 @@ export default Vue.extend({
...
         const is_at_default_diagonals = diagonal_params.every((param) => param?.value === 0.0)
-        results[compass.param] = offset_params.isEmpty() || diagonal_params.isEmpty()
-          || !is_at_default_offsets || !is_at_default_diagonals || scale_param?.value !== 0.0
+        if (is_at_default_diagonals) {
+          results[compass.param] = {
+            calibration: CalibrationType.QUICK,
+            color: 'orange',
+            alert: 'warning',
+            description: 'Quick Calibration is easier for large vehicle, but has the poorer '
+  + 'performance (consider using log-based calibration afterwards).',
```
Comment (ES-Alexander) [inline core/frontend/src/components/vehiclesetup/configuration/compass/ArdupilotMavlinkCompassSetup.vue:357]:
```suggestion
            description: 'Quick Calibration is easier for large vehicles, but has worse '
  + 'performance - consider using log-based calibration afterwards, to refine results.',
```

File: core/frontend/src/components/vehiclesetup/configuration/compass/ArdupilotMavlinkCompassSetup.vue [hunk 4]
```diff
@@ -309,15 +325,64 @@ export default Vue.extend({
...
+        }
+
+        const scale_param_name = `COMPASS_SCALE${compass_number}`
+        const scale_param = autopilot_data.parameter(scale_param_name)
+        const is_at_default_scale = scale_param?.value === 0.0
+
+        if (is_at_default_scale) {
+          results[compass.param] = {
+            calibration: CalibrationType.FULL_NO_WMM,
+            color: 'orange',
```
Comment (ES-Alexander) [inline core/frontend/src/components/vehiclesetup/configuration/compass/ArdupilotMavlinkCompassSetup.vue:370]: Can this use `var(--warning)` from our theme stuff? And similar for the other colours?

File: core/frontend/src/components/vehiclesetup/configuration/compass/ArdupilotMavlinkCompassSetup.vue [hunk 5]
```diff
@@ -309,15 +325,64 @@ export default Vue.extend({
...
+            alert: 'warning',
+            description: 'Full Calibration (No WMM). This means you attempted to calibrate '
+            + 'the compass, but you had no Position data available. This can result in poor performance,'
+            + ' consider retrying with a valid position.',
```
Comment (ES-Alexander) [inline core/frontend/src/components/vehiclesetup/configuration/compass/ArdupilotMavlinkCompassSetup.vue:374]:
```suggestion
            description: 'Calibrated, but without a known (detected or specified) global position, '
            + 'so no corrections were applied from the internal World Magnetic Model (WMM). '
            + 'Consider retrying with a valid position to improve compass performance.',
```

File: core/frontend/src/components/vehiclesetup/configuration/compass/ArdupilotMavlinkCompassSetup.vue [hunk 6]
```diff
@@ -309,15 +325,64 @@ export default Vue.extend({
...
+            calibration_short: 'Calibrated (No WMM)',
+          }
+          continue
+        }
+        results[compass.param] = {
+          calibration: CalibrationType.FULL,
+          color: 'green',
+          alert: 'success',
+          description: 'Fully calibrated, scaled with the internal World Magnetic Model (WMM) database.',
```
Comment (ES-Alexander) [inline core/frontend/src/components/vehiclesetup/configuration/compass/ArdupilotMavlinkCompassSetup.vue:383]:
```suggestion
          description: 'Fully calibrated, including corrections from the internal World Magnetic Model (WMM) database.',
```

**Review by ES-Alexander — CHANGES_REQUESTED** [review]
Nice improvements here :-)

Feedback is around some language changes, and a request not to use hardcoded colours :-)

File: core/frontend/src/components/vehiclesetup/configuration/compass/ArdupilotMavlinkCompassSetup.vue (hunk 4 above)
Comment (Williangalvani) [reply core/frontend/src/components/vehiclesetup/configuration/compass/ArdupilotMavlinkCompassSetup.vue]: done =]

**Review by patrickelectric — APPROVED** [review]

**Review by ES-Alexander — APPROVED** [review]
Changes look good - nice! :-)

**Pull request merged**